src/towncrier/_settings/load.py

```diff
@@ -67,7 +67,7 @@ def load_config_from_options(
 ) -> tuple[str, Config]:
     if config_path is None:
         if directory is None:
-            directory = os.getcwd()
+            return traverse_for_config(None)
```

Comment:
I think that this should be private.

I know that this is from `_settings.load` , but I think that we should make it explicit that we don't want to expose this as a public API.

Also, there are a lot of branches here,
so I think that it would help to add an explicit comment for this branch.

With the missing docstring for the `load_config_from_options` it's a bit hard to understand the purpose of this function.

Can we have something like this.

It lools like a bit of duplication between 
 `traverse_for_config` and these following lines

```python
base_directory = os.path.abspath(directory)
config = load_config(base_directory)
```

maybe have something like this

```python
    if config_path is None:
        return _traverse_for_config(directory)
```

Reply:
I don't think there's any assumption that non underscored methods become automatically part of a public API is there? If people _want_ to shoot themselves in the foot, let em 🤠

Reply:
I've added a docstring and some comments to explain the different branches (I don't like the complexity of the towncrier's loading, but I'm not going to rewrite it just for this addition)

Reply:
Actually, I see what you're getting at with the branching logic now. I've simplified it (and added another test).